src/Our.Umbraco.UnVersion/Services/UnVersionService.cs

```diff
@@ -1,144 +1,207 @@
+                {
+                    var dateRemoveBefore = currentDateTime.AddDays(0 - configEntry.MaxDays);
+                    if (version.UpdateDate < dateRemoveBefore)
+                    {
+                        _logger.Debug<UnVersionService>("Remove version {versionId}, because version is updated {updateDate} and max days is {maxDays} (cutoff: {dateRemoveBefore})", version.VersionId, version.UpdateDate, configEntry.MaxDays, dateRemoveBefore);
+                        versionIdsToDelete.Add(version.VersionId);
+                    }
+                }
+
+            }
+
+            return versionIdsToDelete;
+
+        }
+
+        private List<int> GetNodeIdsFromXpath(string xpath)
+        {
+            using (var ctx = _context.EnsureUmbracoContext())
+            {
+                var nodes = ctx.UmbracoContext.Content.GetByXPath(xpath);
+
+                if (nodes == null)
+                    return new List<int>();
+
+                return nodes.Select(x => x.Id).ToList();
+            }
+        }
+
+        /// <summary>
+        /// Based on configEntries, deletes items from trash if too old or too many.
+        /// </summary>
+        public void CleanUpTrash()
+        {
+            _logger.Debug<CleanUpTrashTask>("CleanupTrashTask PerformRun start");
+
+            using (var scope = _scopeProvider.CreateScope())
+            {
+                using (var umbracoContextReference = _umbracoContextFactory.EnsureUmbracoContext())
+                {
+                    foreach (var cfg in _config.TrashConfigEntries)
+                    {
+                        var query = scope.SqlContext.Query<IContent>();
+
+                        if (cfg.DocTypeAlias != UnVersionConfig.AllDocumentTypesKey)
+                        {
+                            // The query does *not* like it when asked to match ContentType.Alias directly, so we get the ID instead
+                            var contentType = umbracoContextReference.UmbracoContext.Content.GetContentType(cfg.DocTypeAlias);
+
+                            if (contentType != null)
+                            {
+                                query = query.Where(x => x.ContentTypeId.Equals(contentType.Id));
+                            }
+                        }
+
+                        if (cfg.MaxDays < int.MaxValue)
+                        {
+                            // doesn't matter if config says -10 or 10, we always need a negative number
+                            var alwaysNegative = -Math.Abs(cfg.MaxDays); 
+
+                            var minDate = DateTime.Now.AddDays(alwaysNegative);
+
+                            query = query.Where(x => x.UpdateDate < minDate);
+                        }
+
+                        // We'll not handle more than 100 nodes per cycle - if we don't handle it now, we'll get to it eventually
+                        var nodesToDelete = _contentService.GetPagedContentInRecycleBin(

```

Comment:
If the config has a MaxCount, this one should skip the first x (x = MaxCount) nodes before deleting.

I think it can be something like 
```
IEnumerable<IContent> nodesToDelete;
if (cfg.MaxCount > 0)
{
    var i = 1;
    while (true)
    {
        nodesToDelete = _contentService.GetPagedContentInRecycleBin(i, cfg.MaxCount, out long totalRecords);
        if (!nodesToDelete.Any())
        {
            break;
        }
        else
        {
            foreach (var node in nodesToDelete)
            {
                _logger.Debug<CleanUpTrashTask>("Deleting content {contentId} from Recycle Bin", content.Id);
                _contentService.Delete(content);
            }
            i++;
        }
    }
}
else
{
    // just delete everything - this will only be problematic on the first run and if the recycle bin is huge
    nodesToDelete = _contentService.GetPagedContentInRecycleBin(0, int.MaxValue, out long totalRecords);
    foreach (var node in nodesToDelete)
    {
        _logger.Debug<CleanUpTrashTask>("Deleting content {contentId} from Recycle Bin", content.Id);
        _contentService.Delete(content);
    }
}
```